packages/docusaurus-plugin-content-docs/src/globalData.ts

```diff
@@ -24,6 +28,16 @@ export function toGlobalDataDoc(doc: DocMetadata): GlobalDoc {
   };
 }
 
+export function toGlobalDataGeneratedIndex(
+  doc: CategoryGeneratedIndexMetadata,
+): GlobalDoc {
+  return {
+    id: doc.slug,
```

Comment:
This is fine. We never hook into a doc's ID in our built-in hooks, and user can only call hooks on generated index when they swizzle `DocCategoryGeneratedIndex`. At this point, it's already known that the doc you are receiving is a generated index, so using the `id` doesn't make sense.